aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,312 @@
+    // should be safe to list all before returning the list, instead of dynamically load the list.
+    String nextToken = null;
+    List<Namespace> results = new ArrayList<>();
+    while (true) {
+      GetDatabasesResponse response = glue.getDatabases(GetDatabasesRequest.builder()
+          .catalogId(catalogId)
+          .nextToken(nextToken)
+          .build());
+      if (response.hasDatabaseList()) {
+        results.addAll(response.databaseList().stream()
+            .map(GlueToIcebergConverter::toNamespace)
+            .collect(Collectors.toList()));
+        nextToken = response.nextToken();
+      }
+      if (nextToken == null) {
+        break;
+      }
+    }
+    return results;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) throws NoSuchNamespaceException {
+    String databaseName = IcebergToGlueConverter.toDatabaseName(namespace);
+    try {
+      GetDatabaseResponse response = glue.getDatabase(GetDatabaseRequest.builder()
+          .catalogId(catalogId)
+          .name(databaseName)
+          .build());
+      return response.database().parameters();
+    } catch (InvalidInputException e) {
+      throw new NoSuchNamespaceException("invalid input for namespace %s, error message: %s",
+          namespace, e.getMessage());
+    } catch (EntityNotFoundException e) {
+      throw new NoSuchNamespaceException("fail to find Glue database for namespace %s, error message: %s",
+          databaseName, e.getMessage());
+    }
+  }
+
+  /**
+   * After completing this operation, you no longer have access to the tables
+   * (and all table versions and partitions that might belong to the tables)
+   * and the user-defined functions in the deleted database.
+   * AWS Glue deletes these "orphaned" resources asynchronously in a timely manner, at the discretion of the service.
+   * To ensure the immediate deletion of all related resources, before calling DeleteDatabase, use DeleteTableVersion
+   * or BatchDeleteTableVersion, DeletePartition or BatchDeletePartition, DeleteUserDefinedFunction,
+   * and DeleteTable or BatchDeleteTable, to delete any resources that belong to the database.
+   * @param namespace a namespace. {@link Namespace}
+   * @return always true
+   * @throws NamespaceNotEmptyException never thrown
+   */
+  @Override
+  public boolean dropNamespace(Namespace namespace) throws NamespaceNotEmptyException {
+    namespaceExists(namespace);
+    glue.deleteDatabase(DeleteDatabaseRequest.builder()
+        .catalogId(catalogId)
+        .name(IcebergToGlueConverter.toDatabaseName(namespace))
+        .build());
+    // Always successful, otherwise exception is thrown
+    return true;
+  }
+
+  @Override
+  public boolean setProperties(Namespace namespace, Map<String, String> properties) throws NoSuchNamespaceException {
+    Map<String, String> parameter = Maps.newHashMap();
+    parameter.putAll(loadNamespaceMetadata(namespace));
+    parameter.putAll(properties);
```

Comment:
Maybe I'm just totally missing the obvious but I don't see where `parameter` is being used. Loks like it's built and then the database is updated directly via `propeties`.

Reply:
It's used in `persistGlueTable` in `doCommit`

Reply:
Ok. But to me, it seems like the locally declared `parameter` isn't used in this function.

Are you sure you didn't intend to use the merged `parameter` map instead of `properties` here when calling `IcebergToGlueConverter.toDatabaseInput(namespace, properties)`? https://github.com/apache/iceberg/pull/1633/files#diff-51a208168318cb652ad5beb86d3433d714aca916f9816c4cffcc88e273daf6afR287?

Reply:
Conversely, the updated local metadata map is used in `removeProperties` instead of the passed in `properties`. So I still feel like potentially this variable is unused / was potentially meant to be passed into the `IcebergToGlueConverter.toDatabaseInput` instead of passing in the `properties` map.

Here's the usage in `removeProperties` that follows a similar pattern:
- Build an updated table properties map.
- Use the updated map when passing to the iceberg to glue converted for `databaseInput`

https://github.com/apache/iceberg/pull/1633/files#diff-51a208168318cb652ad5beb86d3433d714aca916f9816c4cffcc88e273daf6afR296-R302

Reply:
Oh yeah sorry that one was actually my typo, thank you for pointing out. Somehow I was reading the wrong line when doing the reply...
I have pushed the latest code that has the fix and also with the integration test against actual Glue service that detected those bugs.